src/app/fsat/fsat.component.ts

```diff
@@ -427,4 +424,22 @@ export class FsatComponent implements OnInit {
       });
     });
   }
+
+  initUpdateUnitsModal(oldSettings: Settings) {
+    this.oldSettings = oldSettings;
+    setTimeout(() => {
+    this.showUpdateUnitsModal = true;
+    })
+  }
+
+  closeUpdateUnitsModal() {
+    this.showUpdateUnitsModal = false;
+  }
+
+  selectUpdateAction(shouldUpdateData: boolean) {
+    if(shouldUpdateData == false) {
```

Comment:
If shouldUpdateData == true, we need to convert the input data. See same method in phast.component.ts. If possible, pull out the conversion logic from system-basics --> updateData() and move it to a service method like is done in ConvertPhastService -->convertExistingData(). Otherwise pull the logic out and place in fsat.component.